x-pack/plugins/security_solution_serverless/server/task_manager/nlp_cleanup_task/nlp_cleanup_task.ts

```diff
@@ -0,0 +1,196 @@
+        schedule: {
+          interval: NLPCleanupTaskConstants.INTERVAL,
+        },
+        state: {},
+        params: { version: NLPCleanupTaskConstants.VERSION },
+      });
+
+      if (this.productTier !== ProductTier.complete) {
+        // In case the task was previously ran and is scheduled for later in the future, run now
+        this.logger.info('Scheduling for immediate run');
+        await taskManager.runSoon(this.taskId);
+      }
+    } catch (e) {
+      this.logger.error(
+        `Error scheduling task ${NLPCleanupTaskConstants.TYPE}, received ${e.message}`
+      );
+    }
+  };
+
+  private runTask = async (taskInstance: ConcreteTaskInstance, core: CoreSetup) => {
+    // if task was not `.start()`'d yet, then exit
+    if (!this.wasStarted) {
+      this.logger.debug('[runTask()] Aborted. Task not started yet');
+      return;
+    }
+
+    // Check that this task is current
+    if (taskInstance.id !== this.taskId) {
+      // old task, return
+      throwUnrecoverableError(new Error('Outdated task version'));
+    }
+
+    const [{ elasticsearch }] = await core.getStartServices();
+    const esClient = elasticsearch.client.asInternalUser;
+
+    try {
+      const trainedModels = await esClient.ml.getTrainedModels();
+      const pytorchModels = trainedModels.trained_model_configs.filter(
+        (config) => config.model_type === 'pytorch'
+      );
+
+      if (pytorchModels.length === 0) {
+        this.logger.info('No pytorch models found, exiting task');
+        return { state: {} };
+      }
+
+      this.logger.info(
+        `[${pytorchModels.length}] pytorch model(s) found. Stopping, then deleting models/aliases`
+      );
+
+      // First attempt to stop all models, use force=true to ensure any linked pipelines don't get in the way
+      const stopResponses = await Promise.all(
+        pytorchModels.map((m) =>
+          esClient.ml.stopTrainedModelDeployment({ model_id: m.model_id, force: true })
+        )
+      );
+      const successfulStops = stopResponses.filter((r) => r.stopped).length;
+      const failedStops = stopResponses.length - successfulStops;
+      this.logger.info(
+        `[${successfulStops}] model(s) successfully stopped, [${failedStops}] model(s) failed to stop`
+      );
+
+      // Then delete all models, use force=true again to ensure any linked pipelines don't get in the way
+      const deleteResponses = await Promise.all(
+        pytorchModels.map((m) =>
+          esClient.ml.deleteTrainedModel({ model_id: m.model_id, force: true })
+        )
+      );
+      const successfulDeletes = deleteResponses.filter((r) => r.acknowledged).length;
+      const failedDeletes = deleteResponses.length - successfulDeletes;
+      this.logger.info(
+        `[${successfulDeletes}] model(s) successfully deleted, [${failedDeletes}] model(s) failed to delete`
+      );
+
+      // TODO: Any manual cleanup around aliases? They appear to be deleted with the model...
```

Comment:
To the ML folks -- [it was mentioned](https://github.com/elastic/security-team/issues/7995#issuecomment-1911967874) that aliases would need to be cleaned up separately. In my testing I'm seeing that force deleting the model automatically cleans up any aliases. So question here is if any explicit alias cleanup is necessary. Thanks!